File: src/main/java/com/kingg/api_vacunas_panama/persistence/entity/RolesPermisosId.java
package com.kingg.api_vacunas_panama.persistence.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.Hibernate;

import java.io.Serial;
import java.io.Serializable;
import java.util.Objects;

@Embeddable
@NoArgsConstructor
@Getter
@Setter
public class RolesPermisosId implements Serializable {
    @Serial
    private static final long serialVersionUID = -5300599546506914322L;
    @NotNull
    @Column(name = "rol", nullable = false)
    private Short idRol;

    @NotNull
    @Column(name = "permiso", nullable = false)
    private Short idPermiso;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        RolesPermisosId entity = (RolesPermisosId) o;
        return Objects.equals(this.idRol, entity.idRol) &&
                Objects.equals(this.idPermiso, entity.idPermiso);
    }

    @Override
    public int hashCode() {
        return Objects.hash(idRol, idPermiso);
    }

}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/persistence/entity/RolesPermisosId.kt
package io.github.kingg22.api.vacunas.panama.persistence.entity

import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import org.hibernate.Hibernate
import java.io.Serial
import java.io.Serializable
import java.util.Objects

@Embeddable
class RolesPermisosId(
    @Column(name = "rol", nullable = false)
    var idRol: Short,

    @Column(name = "permiso", nullable = false)
    var idPermiso: Short,
) : Serializable {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        val entity = other as RolesPermisosId
        return this.idRol == entity.idRol && this.idPermiso == entity.idPermiso
    }

    override fun hashCode(): Int = Objects.hash(idRol, idPermiso)

    companion object {
        @Serial
        @JvmStatic
        private val serialVersionUID = -5300599546506914322L
    }
}
